src/test/java/uk/gov/justice/hmpps/prison/service/OffenderAssessmentServiceTest.java

```diff
@@ -402,6 +402,50 @@ public void getCurrentCsraClassification_returnsNullIfNoAssessments() {
         assertThat(csraClassificationCode).isEqualTo(null);
     }
 
+    @Test
+    public void getOffendersAssessmentRatings_returnsResultsOfNextAssessmentIfFirstNotSet() {
+        when(repository.findByCsraAssessmentAndByOffenderNo_OrderByLatestFirst("N1234AA")).thenReturn(List.of(
+            getOffenderAssessment_CsraClassificationBuilder(null, LocalDate.parse("2019-01-03"))
+                .build(),
+            getOffenderAssessment_CsraClassificationBuilder(new AssessmentClassification("HI", "High"), LocalDate.parse("2019-01-02"))
+                .build()
+        ));
+
+        final var csraRatings = service.getOffendersAssessmentRatings(List.of("N1234AA"));
+
+        assertThat(csraRatings).isEqualTo(List.of(
+            uk.gov.justice.hmpps.prison.api.model.AssessmentClassification.builder()
+                .offenderNo("N1234AA")
+                .classificationCode("HI")
+                .classificationDate(LocalDate.parse("2019-01-02"))
+                .build()
+            )
+        );
+    }
+
+    @Test
+    public void getOffendersAssessmentRatings_returnsAllOffendersRequestedEvenIfNoAssessments() {
+        when(repository.findByCsraAssessmentAndByOffenderNo_OrderByLatestFirst("N1234AA")).thenReturn(List.of(
+            getOffenderAssessment_CsraClassificationBuilder(new AssessmentClassification("HI", "High"), LocalDate.parse("2019-01-02"))
+            .build()
+        ));
+        when(repository.findByCsraAssessmentAndByOffenderNo_OrderByLatestFirst("N2345BB")).thenReturn(List.of());
+
+        final var csraRatings = service.getOffendersAssessmentRatings(List.of("N1234AA", "N2345BB"));
+
+        assertThat(csraRatings).isEqualTo(List.of(
+            uk.gov.justice.hmpps.prison.api.model.AssessmentClassification.builder()
```

Comment:
Adding a comment here in case you missed the absolute type reference.

Reply:
I didn't - there is a name clash, but I couldn't think of a better name so i just used the full type ref